src/gui/owncloudgui.cpp

```diff
@@ -51,61 +53,51 @@ namespace {
 {
-    auto account = accountStatePtr->account();
-
-    // ensure we are connected and fetch the capabilities
-    auto validator = new ConnectionValidator(account, account.data());
-
-    QObject::connect(validator, &ConnectionValidator::connectionResult, account.data(), [accountStatePtr, localFolder, syncMode](ConnectionValidator::Status status, const QStringList &errors) {
-        if (OC_ENSURE(status == ConnectionValidator::Connected)) {
-            // saving once after adding makes sure the account is stored in the config in a working state
-            // this is needed to ensure a consistent state in the config file upon unexpected terminations of the client
-            // (for instance, when running from a debugger and stopping the process from there)
-            AccountManager::instance()->save(true);
-            auto folderMan = FolderMan::instance();
-
-            // saves a bit of duplicate code
-            auto addFolder = [folderMan, accountStatePtr, syncMode](const QString &localFolder, const QString &remotePath, const QUrl &webDavUrl, const QString &displayName = {}) {
-                folderMan->addFolderFromWizard(accountStatePtr, localFolder, remotePath, webDavUrl, displayName, syncMode);
-            };
-
-            auto finalize = [accountStatePtr] {
-                accountStatePtr->checkConnectivity();
-                FolderMan::instance()->scheduleAllFolders();
-            };
-
-            if (accountStatePtr->account()->capabilities().spacesSupport().enabled) {
-                auto *drive = new OCC::GraphApi::Drives(accountStatePtr->account());
-
-                QObject::connect(drive, &OCC::GraphApi::Drives::finishedSignal, [accountStatePtr, localFolder, drive, addFolder, finalize] {
-                    if (drive->parseError().error == QJsonParseError::NoError) {
-                        const auto &drives = drive->drives();
-                        if (!drives.isEmpty()) {
-                            const QDir localDir(localFolder);
-                            localDir.mkdir(".");
-                            for (const auto &d : drives) {
-                                const QDir driveLocalFolder = localDir.filePath(d.getName());
-                                driveLocalFolder.mkdir(".");
-                                addFolder(driveLocalFolder.absolutePath(), {}, QUrl::fromEncoded(d.getRoot().getWebDavUrl().toUtf8()), d.getName());
-                            }
-                            finalize();
-                        }
-                    }
-                });
+    // saving once after adding makes sure the account is stored in the config in a working state
+    // this is needed to ensure a consistent state in the config file upon unexpected terminations of the client
+    // (for instance, when running from a debugger and stopping the process from there)
+    AccountManager::instance()->save(true);
```

Comment:
I guess that should happen in runNewAccountWizard after the connection validator?

Reply:
Perhaps it makes sense to save the account even before running the validator. What if client crashes in the validator step? The account might be stored in an inconsistent state then, too.